spec/requests/shopify/export_product_spec.rb

```diff
@@ -0,0 +1,62 @@
+require 'spec_helper'
+require 'active_resource/base_decorator'
+
+describe 'Export a Spree Product to Shopify' do
+  include_context 'shopify_exporter_helpers'
+  include_context 'shopify_helpers'
+
+  let(:spree_product) { create(:product, name: 'Product Name') }
+  subject { find_shopify_product(spree_product) }
+
+  it 'creates a new product' do
+    export_product!(spree_product)
+    expect(subject).to be_truthy
+    cleanup_shopify_product!(subject)
+  end
+
+  describe 'when the product existed on Shopify but was deleted' do
+    let!(:existing_product) { export_product!(spree_product) }
+
+    before do
+      shopify_product = find_shopify_product(spree_product)
+      cleanup_shopify_product!(shopify_product)
+    end
+
+    it 'creates a new product' do
+      export_product!(spree_product)
+      expect(subject).to be_truthy
+    end
+
+    after do
```

Comment:
What do you think about moving this up to right after the before do block? Therefore the reader has all the context required to understand the following specs?